config/database_test.go

```diff
@@ -989,7 +1054,7 @@ func TestDatabaseStoreString(t *testing.T) {
 	_, tearDown := setupConfigDatabase(t, emptyConfig, nil)
 	defer tearDown()
 
-	ds, err := newTestDatabaseStore(t)
+	ds, err := newTestDatabaseStore(t, nil)
```

Comment:
What happens if the env. variable for the custome defaults is set but the file is not in place?

Reply:
Good question. This case is not contemplated at the `ConfigStore` creation level as we're passing the config struct to it, but managed [as part of the server command](https://github.com/mattermost/mattermost-server/pull/16265/files#diff-206ba9b9ee4ca76d959e08e934151f296c42002c0a03d50f44c64ecfdcdd6c85R46), and the command will fail if the file is not present or cannot be opened/correctly decoded.

I'm actually 0/5 on this, the server could just try to load the file and simply log an error if any and continue without the custom defaults, but my understanding is that we would prefer for the pod not to start if the custom defaults variable is present but cannot be loaded.